src/main/java/org/opensearch/knn/index/ModelContext.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.ToXContentFragment;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.index.mapper.MapperParsingException;
+import org.opensearch.knn.index.util.KNNEngine;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.opensearch.knn.common.KNNConstants.KNN_ENGINE;
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_SPACE_TYPE;
+import static org.opensearch.knn.common.KNNConstants.MODEL_ID;
+
+/**
+ * Used to capture context of a model
+ */
+public class ModelContext implements ToXContentFragment {
+
+    private final String modelId;
+    private final KNNEngine knnEngine;
+    private final SpaceType spaceType;
+
+    /**
+     * Constructor
+     *
+     * @param modelId Identifier for model
+     * @param knnEngine KNNEngine of model
+     * @param spaceType SpaceType of model
+     */
+    public ModelContext(String modelId, KNNEngine knnEngine, SpaceType spaceType) {
+        this.modelId = modelId;
+        this.knnEngine = knnEngine;
+        this.spaceType = spaceType;
+    }
+
+    /**
+     * Getter for model's id
+     *
+     * @return model's id
+     */
+    public String getModelId() {
+        return modelId;
+    }
+
+    /**
+     * Getter for model's engine
+     *
+     * @return model's engine
+     */
+    public KNNEngine getKNNEngine() {
+        return knnEngine;
+    }
+
+    /**
+     * Getter for model's space type
+     *
+     * @return model's space type
+     */
+    public SpaceType getSpaceType() {
+        return spaceType;
+    }
+
+    /**
+     * Parse an object to a ModelContext
+     *
+     * @param in String of model id
+     * @return ModelContext constructed from model identified by in
+     */
+    public static ModelContext parse(Object in) {
+        if (!(in instanceof Map)) {
+            throw new MapperParsingException("Unable to parse ModelContext: provided input is not of type \"Map\"");
+        }
+
+        @SuppressWarnings("unchecked")
+        Map<String, Object> methodMap = (Map<String, Object>) in;
+
+        String modelId = null;
+        KNNEngine knnEngine = null;
+        SpaceType spaceType = null;
+
+        String key;
+        Object value;
+        for (Map.Entry<String, Object> methodEntry : methodMap.entrySet()) {
+            key = methodEntry.getKey();
+            value = methodEntry.getValue();
+
+            if (MODEL_ID.equals(key)) {
```

Comment:
Should we do ```equalsIgnoreCase``` to avoid case sensitive issues? Same comments for all the places we are doing String equality check.

Reply:
So, IMHO, I like to stick with one unique identifier for field variables. This keeps the interface consistent.

In OpenSearch, the type is case sensitive. For example, this fails:
```
curl -X PUT "localhost:9200/testindex?pretty" -H 'Content-Type: application/json' -d'
{
  "mappings": {
    "properties": {
      "testfield": {
        "type": "Text"
      }
    }
  }
}
'
```

@vamshin if you disagree, we can have @VijayanB be the tie breaker.

Reply:
Agree its good to keep it consistent. Please ignore my comment. Thanks